src/node/internal/crypto_keys.ts

```diff
@@ -276,101 +338,112 @@ export function createSecretKey(
-//     if (key.type !== 'private') {
-//       throw new ERR_INVALID_ARG_VALUE('key', key, 'must be a private key');
-//     }
-//     inner.key = (key as KeyObject)[kHandle];
-//   } else if (type === kPublicKey && key instanceof CryptoKey) {
-//     // Covers deriving public key from a private key.
-//     if ((key as CryptoKey).type !== 'private') {
-//       throw new ERR_INVALID_ARG_VALUE('key', key, 'must be a private key');
-//     }
-//     inner.key = key as CryptoKey;
-//   } else {
-//     const options = key as CreateAsymmetricKeyOptions;
-//     if (typeof options.key === 'string') {
-//       inner.key = Buffer.from(options.key as string, options.encoding);
-//     } else if (isArrayBufferView(options.key)) {
-//       inner.key = options.key as ArrayBufferView;
-//     } else if (isArrayBuffer(options.key)) {
-//       inner.key = options.key as ArrayBuffer;
-//     } else if (isSharedArrayBuffer(options.key)) {
-//       inner.key = options.key as SharedArrayBuffer;
-//     } else if (type === kPublicKey && key instanceof KeyObject) {
-//       if ((options.key as KeyObject).type !== 'private') {
-//         throw new ERR_INVALID_ARG_VALUE('options.key', options.key, 'must be a private key');
-//       }
-//       inner.key = (options.key as KeyObject)[kHandle];
-//     } else if (type === kPublicKey && key instanceof CryptoKey) {
-//       if ((options.key as CryptoKey).type !== 'private') {
-//         throw new ERR_INVALID_ARG_VALUE('options.key', options.key, 'must be a private key');
-//       }
-//       inner.key = options.key as CryptoKey;
-//     } else {
-//       inner.key = key as JsonWebKey;
-//     }
-//     validateKeyData(inner.key, 'options.key', { allowObject: true });
-
-//     if (options.format !== undefined) {
-//       validateString(options.format, 'options.format');
-//       inner.format = options.format;
-//     }
-//     if (options.type !== undefined) {
-//       validateString(options.type, 'options.type');
-//       inner.type = options.type;
-//     }
-//     if (options.passphrase !== undefined) {
-//       if (typeof options.passphrase === 'string') {
-//         inner.passphrase = Buffer.from(options.passphrase, options.encoding);
-//       } else {
-//         if (!isUint8Array(options.passphrase)) {
-//           throw new ERR_INVALID_ARG_TYPE('options.passphrase', [
-//             'string', 'Uint8Array'
-//           ], options.passphrase);
-//         }
-//         inner.passphrase = options.passphrase;
-//       }
-//       if (inner.passphrase.byteLength > 1024) {
-//         throw new ERR_INVALID_ARG_VALUE('options.passphrase', options.passphrase.length, '<= 1024');
-//       }
-//     }
-//   }
-//   return inner;
-// }
+function prepareAsymmetricKey(
+  key: CreateAsymmetricKeyOptions | KeyData,
+  ctx: KeyContext
+): InnerCreateAsymmetricKeyOptions {
+  // Safety check... key should not be undefined or null here.
+  if ((key as any) == null) {
```

Comment:
If we are doing this, we should change key type...

Reply:
This is intentional. Even when using typescript it is still possible for the value to be passed as null/undefined at runtime so this check adds an additional fallback.

Reply:
if you change line 388 to `| null | undefined`, we won't need this as any check